EIPS/eip-7885.md

```diff
@@ -0,0 +1,217 @@
+
+## Rationale
+
+If $f$ and $g$ are two polynomials of $R$, then
+$f\times g= \text{NTT\_INV}(\text{NTT\_VECMULMOD}(
+\text{NTT\_FW}(a), \text{NTT\_FW}(b)))$ is equal to the product of $f$ and $g$ in $R$. The algorithm has a complexity of $n \log_2n$ rather than $n^2$ with the classical schoolbook multiplication algorithm.
+
+### Fields of interest
+
+The implementation applies for many fields of interest for cryptography. In particular, the design applies for:
+
+- FALCON: $q=3.2^{12}+1$ (one of the NIST winners for  post-quantum signature scheme),
+
+- DILITHIUM: $q=2^{23}-2^{13}+1$ (one of the NIST winners for post-quantum signature scheme),
+
+- KYBER: $q=13.2^8+1$ (one of the NIST winners for post-quantum key encapsulation mechanism),
+
+- Babybear: $q=15.2^{27}+1$ (Risc0),
+
+- Goldilocks: $q=2^{64}-2^{32}+1$ (Polygon's Plonky2),
+
+- M31: $q=2^{31}-1$ (Circle STARKS, STwo, Plonky3),
+
+- StarkCurve: $q=2^{251}+17.2^{192}+1$
+
+
+### Benchmarks
+
+#### Pure solidity
+
+To illustrate the interest of the precompile, the assets provide the measured gas const for a single NTT and extrapolates the minimal gas cost taking into account the required number of NTT_FW and NTT_INV. The provided assets use pure Yul optimizations, with memory access hacks. It is unlikely that more than one order of magnitude could be spared on such a minimal code. 
+
+|Use case| Parameters                   | single NTT gas cost         |  Required NTT(FW/INV)    | Estimated NTT/Full cost |
+|--|------------------------|---------------------|---------------------|---|
+|Falcon| $q=12289, n=512$       | 1.8 M | 1 NTTFW+1 NTTINV |3.6 M| 
+|Dilithium| $q=2^{23}-2^{13}+1, n=256$| 460K | 4 NTTFW +1 NTTINV|2.3M|
+
+Falcon cost has been measured over a full implementation and is compliant to the estimation. Dilithium cost is evaluated assuming
+
+This demonstrates that using pure solidity enables L2s with low gas fees to experiment with FALCON in the short term, whereas it is too expensive to do so on L1.
+Adopting this EIP, the signature verification of Falcon can be reduced to **1500** gas, and a similar result is expected for Dilithium.
+Adopting the hash function as a separate EIP would enable a gas verification cost of 2000 gas.
+This is in line with the ratio looking at SUPERCOP implementations.
+
+
+
+
+## Backwards Compatibility
+
+There are no backward compatibility questions.
+
+## Test Cases
+
+There are no edge cases in the considered operations.
+
+
+## Reference Implementation
+
+
+There are two fully spec compatible implementations on the day of writing:
+
+- a python reference code provided in the assets of this EIP
+
+- a solidity reference code provided in the assets of this EIP
+
+Both codes have been validated over a large base of reference vectors, and implementing both FALCON and DILITHIUM algorithms as demonstration of the usefulness of the precompile.
+
+
+## Security Considerations
+
+Needs discussion.
```

Comment:
add  <-- TODO --> html marker here that bots can flag for you when you attempt to move EIP post draft